prov/efa/src/rdm/efa_rdm_ep.h

```diff
@@ -244,7 +241,7 @@ static inline size_t efa_rdm_ep_get_tx_pool_size(struct efa_rdm_ep *ep)
 
 static inline int efa_rdm_ep_need_sas(struct efa_rdm_ep *ep)
 {
-	return ep->msg_order & FI_ORDER_SAS;
+	return ep->user_info->tx_attr->msg_order & FI_ORDER_SAS;
```

Comment:
probably we want to use `(ep->user_info->tx_attr->msg_order & FI_ORDER_SAS) || (ep->user_info->rx_attr->msg_order & FI_ORDER_SAS)`

Reply:
But AFAICT some applications like MPICH only set SAS for tx_attr.

Reply:
Updated